gradle-baseline-java/src/main/groovy/com/palantir/baseline/tasks/CheckClassUniquenessTask.java

```diff
@@ -23,55 +23,60 @@
 import java.nio.file.Paths;
 import java.util.Collection;
 import java.util.Comparator;
+import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.ModuleVersionIdentifier;
+import org.gradle.api.provider.Provider;
 import org.gradle.api.tasks.InputFiles;
 import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 
 public class CheckClassUniquenessTask extends DefaultTask {
 
-    private Configuration configuration;
+    private Provider<List<Configuration>> configurations;
 
     public CheckClassUniquenessTask() {
         setGroup("Verification");
         setDescription("Checks that the given configuration contains no identically named classes.");
     }
 
     @InputFiles
-    public final Configuration getConfiguration() {
-        return configuration;
+    public final Provider<List<Configuration>> getConfigurations() {
+        return configurations;
     }
 
-    public final void setConfiguration(Configuration configuration) {
-        this.configuration = configuration;
+    public final void setConfigurations(Provider<List<Configuration>> configs) {
+        this.configurations = configs;
```

Comment:
I think the private `configurations` field should really be a ListProperty<Configuration> and call .set() here.